### PR TITLE
Add --all flag for listing all projects, active and inactive

### PR DESCRIPTION
**Pull request by valyo**
> **Before submitting the PR, please go through the sections below and fill in what you can. If there are any items that are irrelevant for the current PR, remove the row. If a relevant option is missing, please add it as an item and add a PR comment informing that the new option should be included into this template.**

> **All _relevant_ items should be ticked before the PR is merged**

Currently Deleted and Archived projects are included when running dds ls or  dds project ls 
Here we limit it to only including the active ones and adding a flag called --all to indicate that the dds should list all of them.
Works with https://github.com/ScilifelabDataCentre/dds_web/pull/1302

- [x] Add a summary of the changes and the related issue
- [x] Add motivation and context regarding why the change is needed
- [x] List / describe any dependencies or other changes required for this change
- [x] Fixes DDS-1306

## Type of change

- [x] New feature (non-breaking)

# Checklist:

## General

- [x] [Changelog](../CHANGELOG.md): New row added. Not needed when PR includes _only_ tests.
- [x] Code change
  - [x] Self-review of code done
  - [x] Comments added, particularly in hard-to-understand areas

## Repository / Releases

- [x] Rebase / update of branch done

## Checks

- [x] Formatting: Black & Prettier checks pass
- [x] CodeQL passes
- [x] Tests
  - [x] The tests pass

### CODE REVIEW

**Comment by inaod568** [comment]
Accidentally wrote this comment in the API PR but: https://github.com/ScilifelabDataCentre/dds_web/pull/1302#discussion_r994687329

File: dds_cli/__main__.py [hunk 1]
```diff
@@ -192,9 +192,10 @@ def dds_main(click_ctx, verbose, log_file, no_prompt, token_path):
 @usage_flag(help_message="Show the usage for available projects, in GBHours and cost.")
 @users_flag(help_message="Display users associated with a project(Requires a project id).")
 @click.option("--projects", "-lp", is_flag=True, help="List all project connected to your account.")
+@click.option("--all", is_flag=True, help="List all project connected to your account.")
```
Comment (inaod568) [inline dds_cli/__main__.py:195]: Maybe this instead to not override the builtin `all`? 
```suggestion
@click.option("--include-inactive", is_flag=True, help="List all project connected to your account.")
```
or 
```suggestion
@click.option("--show-all", is_flag=True, help="List all project connected to your account.")
```

File: dds_cli/__main__.py [hunk 2]
```diff
@@ -884,13 +886,14 @@ def project_group_command(_):
 # Flags
 @usage_flag(help_message="Show the usage for available projects, in GBHours and cost.")
 @json_flag(help_message="Output project list as json.")  # users, json, tree
+@click.option("--all", is_flag=True, help="List all project connected to your account.")
```
Comment (inaod568) [inline dds_cli/__main__.py:889]: Same here

File: dds_cli/data_lister.py [hunk 3]
```diff
@@ -95,7 +97,8 @@ def list_projects(self, sort_by="Updated"):
         always_show = response.get("always_show", False)
         if not project_info:
             raise exceptions.NoDataError("No project info was retrieved. No files to list.")
-
+        if not self.all:
+            project_info = [p for p in project_info if p["Is Active"]]
```
Comment (inaod568) [inline dds_cli/data_lister.py:101]: Remove this
```suggestion
```
and add this to the method name etc
```
def list_projects(self, sort_by="Updated", <new option name here>: bool = False):
```
and then 
```
        response, _ = dds_cli.utils.perform_request(
            # ... stuff here
            json={"usage": self.show_usage, "get_all or <new option name>": <new option name here>},
            # ... rest of stuff here
        )
```

File: dds_cli/data_lister.py [hunk 4]
```diff
@@ -51,6 +51,7 @@ def __init__(
         tree: bool = False,
         no_prompt: bool = False,
         json: bool = False,
+        all: bool = False,
```
Comment (inaod568) [inline dds_cli/data_lister.py:54]:
```suggestion
```

File: dds_cli/data_lister.py [hunk 5]
```diff
@@ -72,6 +73,7 @@ def __init__(
         self.show_usage = show_usage
         self.tree = tree
         self.json = json
+        self.all = all
```
Comment (inaod568) [inline dds_cli/data_lister.py:76]:
```suggestion
```

**Review by inaod568 — CHANGES_REQUESTED** [review]

File: dds_cli/__main__.py [hunk 6]
```diff
@@ -884,13 +886,14 @@ def project_group_command(_):
 # Flags
 @usage_flag(help_message="Show the usage for available projects, in GBHours and cost.")
 @json_flag(help_message="Output project list as json.")  # users, json, tree
+@click.option("--show-all", is_flag=True, help="List all project connected to your account.")
```
Comment (inaod568) [inline dds_cli/__main__.py:889]:
```suggestion
@click.option("--show-all", is_flag=True, help="List all projects connected to your account (including inactive).")
```

File: dds_cli/__main__.py [hunk 7]
```diff
@@ -192,9 +192,10 @@ def dds_main(click_ctx, verbose, log_file, no_prompt, token_path):
 @usage_flag(help_message="Show the usage for available projects, in GBHours and cost.")
 @users_flag(help_message="Display users associated with a project(Requires a project id).")
 @click.option("--projects", "-lp", is_flag=True, help="List all project connected to your account.")
+@click.option("--show-all", is_flag=True, help="List all project connected to your account.")
```
Comment (inaod568) [inline dds_cli/__main__.py:195]:
```suggestion
@click.option("--show-all", is_flag=True, help="List all projects connected to your account (including inactive).")
```

File: dds_cli/data_lister.py [hunk 8]
```diff
@@ -72,18 +73,19 @@ def __init__(
...
-    def list_projects(self, sort_by="Updated"):
+    def list_projects(self, sort_by="Updated", show_all: bool = False):
         """Get a list of project(s) the user is involved in."""
         # Get projects from API
         response, _ = dds_cli.utils.perform_request(
             DDSEndpoint.LIST_PROJ,
             headers=self.token,
             method="get",
-            json={"usage": self.show_usage},
+            json={"usage": self.show_usage, "show_all": self.show_all},
```
Comment (inaod568) [inline dds_cli/data_lister.py:88]:
```suggestion
            json={"usage": self.show_usage, "show_all": show_all},
```

File: dds_cli/data_lister.py [hunk 9]
```diff
@@ -72,18 +73,19 @@ def __init__(
         self.show_usage = show_usage
         self.tree = tree
         self.json = json
+        self.show_all = show_all
```
Comment (inaod568) [inline dds_cli/data_lister.py:76]:
```suggestion
```

File: dds_cli/data_lister.py [hunk 10]
```diff
@@ -51,6 +51,7 @@ def __init__(
         tree: bool = False,
         no_prompt: bool = False,
         json: bool = False,
+        show_all: bool = False,
```
Comment (inaod568) [inline dds_cli/data_lister.py:54]:
```suggestion
```

File: dds_cli/__main__.py [hunk 11]
```diff
@@ -211,6 +212,7 @@ def list_projects_and_contents(
                 show_usage=usage,
                 no_prompt=click_ctx.get("NO_PROMPT", False),
                 json=json,
+                show_all=show_all,
```
Comment (inaod568) [inline dds_cli/__main__.py:215]: Move this so that it says `projects = lister.list_projects(sort_by=sort, show_all=show_all)` instead
```suggestion
```

**Review by inaod568 — CHANGES_REQUESTED** [review]

File: dds_cli/__main__.py [hunk 12]
```diff
@@ -211,10 +218,11 @@ def list_projects_and_contents(
                 show_usage=usage,
                 no_prompt=click_ctx.get("NO_PROMPT", False),
                 json=json,
+                # show_all=show_all,
```
Comment (inaod568) [inline dds_cli/__main__.py:221]:
```suggestion
```

File: dds_cli/data_lister.py [hunk 13]
```diff
@@ -51,6 +51,7 @@ def __init__(
         tree: bool = False,
         no_prompt: bool = False,
         json: bool = False,
+        # show_all: bool = False,
```
Comment (inaod568) [inline dds_cli/data_lister.py:54]:
```suggestion
```

**Review by inaod568 — CHANGES_REQUESTED** [review]
Looks good! Just remove the commented out mentions of the `show_all`

**Comment by inaod568** [comment]
@valyo Also rebase / merge dev into this branch

**Comment by valyo** [comment]
of course

**Review by inaod568 — APPROVED** [review]
Looks good and works as expected!

**Pull request merged**